### PR TITLE
Reinstate Log Levels and Original Functionality

### PR DESCRIPTION
**Pull request by williams-jack**
## Feature/Problem Description
With deployment chaos _mostly_ over, we now need to reinstate logging level rules, as well as the handling of tempfiles and .tgz image files. We also need to remove exposition of the API endpoint that leaks vulnerable job info. 

## Solution (Changes Made)
* Log level back to being dependent on environment.
* Tempfiles removed on exit of the `with` context.
* TGZ files removed after loading.
  * Function for retrieval of tgz file takes in an images _endpoint_, allowing CI to run tests properly again.

## Additional Notes
* Caught bug where the updated DB ID for a job that has been re-enqueued was not correctly passed to the function that notifies the client of a job's new location; this has been fixed by ensuring it is passed in.

### CODE REVIEW

File: worker/orca_grader/__main__.py [hunk 1]
```diff
@@ -160,8 +161,7 @@ def run_grading_job(grading_job: GradingJobJSON, no_container: bool,
 # TODO: Would it be more useful to return the result of the job here?
 def handle_grading_job(grading_job: GradingJobJSON, image_name: str | None = None,
                        container_cmd: List[str] | None = None) -> None:
-    with tempfile.NamedTemporaryFile(mode="w", suffix='.json',
-                                     delete_on_close=False) as temp_job_file:
+    with tempfile.NamedTemporaryFile(mode="w", suffix='.json') as temp_job_file:
```
Comment (blerner) [inline worker/orca_grader/__main__.py:164]: This change worries me -- I'm not convinced that we'll be able to `docker cp` the tempfile if we do this.  Keep in mind that we still have delete=True, so the file still will be deleted at the end of the with-statement...

File: worker/orca_grader/__main__.py [hunk 2]
```diff
@@ -219,15 +218,16 @@ def handle_grading_job(grading_job: GradingJobJSON, image_name: str | None = Non
         else:
             warn_str = "The processes timed out during execution." if \
                         result.did_timeout else "An error was encountered during execution."
-            _LOGGER.warn(warn_str)
+            _LOGGER.warning(warn_str)
```
Comment (blerner) [inline worker/orca_grader/__main__.py:221]: why the rename here?  (did it work before?  should it have been `warning` all along?)

File: worker/orca_grader/__main__.py [hunk 3]
```diff
@@ -219,15 +218,16 @@ def handle_grading_job(grading_job: GradingJobJSON, image_name: str | None = Non
...
         _LOGGER.debug("\n".join(result.results))
 
 
 def inform_client_of_reenqueue(grading_job: GradingJobJSON,
                                updated_db_id: int) -> None:
     post_job_status_to_client(location="Queue",
                               response_url=grading_job["response_url"],
-                              key=grading_job["key"])
+                              key=grading_job["key"],
+                              id=updated_db_id)
```
Comment (blerner) [inline worker/orca_grader/__main__.py:230]: This seems unrelated to this PR's purpose -- was this just a bug?

File: worker/orca_grader/container/do_grading.py [hunk 4]
```diff
@@ -2,7 +2,6 @@
 import os
 import sys
 import shutil
```
Comment (blerner) [inline worker/orca_grader/container/do_grading.py:4]: I don't think we need shutil; I added it for the unsuccessful debugging attempts

File: worker/orca_grader/__main__.py [hunk 5]
```diff
@@ -170,7 +170,6 @@ def handle_grading_job(grading_job: GradingJobJSON, image_name: str | None = Non
         # Will need to see source code for json.dump.
         temp_job_file.write(json.dumps(grading_job, default=str))
         temp_job_file.flush()
-        temp_job_file.close()
```
Comment (blerner) [inline worker/orca_grader/__main__.py:173]: Same as change above; I think we need this line for the successful docker cp command.

File: worker/orca_grader/__main__.py (hunk 2 above)
Comment (williams-jack) [reply worker/orca_grader/__main__.py:221]: It's a stupid deprecation thing (e.g., the `exists?` vs `exist?` thing with Ruby's `Dir`). Figured I'd get ahead of it

File: worker/orca_grader/__main__.py (hunk 3 above)
Comment (williams-jack) [reply worker/orca_grader/__main__.py:230]: Yup -- this needed to be passed in  to communicate the ID appropriately to bottlenose so it knew how to get the proper status

File: worker/orca_grader/__main__.py (hunk 1 above)
Comment (williams-jack) [reply worker/orca_grader/__main__.py:164]: Fair point, realized I had misinterpreted the doc for `delete` vs `delete_on_close`; fixed now!

File: worker/orca_grader/__main__.py (hunk 3 above)
Comment (williams-jack) [reply worker/orca_grader/__main__.py:230]: I've made a note of this in the PR description in case we need to revisit it later
